# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/PMediaTransport.ipdl
# Commit: 8196045d38b5
# Full Hash: 8196045d38b50e04313725215c36ba1618c50486
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-09-18 21:49:27
# Regressor Bug: 1569183
# File Overlap Count: 2
# Description:
#   Bug 1569183: Create an ipdl struct called WebrtcProxyConfig, to cut down on the arg count on lots of functions. r=mjf,mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45562
# ==============================================================================

diff -r 6e1eba5c88ad -r 8196045d38b5 dom/media/webrtc/PMediaTransport.ipdl
--- a/dom/media/webrtc/PMediaTransport.ipdl	Wed Sep 18 14:27:42 2019 +0000
+++ b/dom/media/webrtc/PMediaTransport.ipdl	Tue Sep 17 18:19:04 2019 +0000
@@ -5,7 +5,7 @@
 include protocol PBackground;
 
 #ifdef MOZ_WEBRTC
-include NeckoChannelParams;
+include WebrtcProxyConfig;
 
 // ParamTraits stuff for generated code and other classes we don't want to change
 include "mozilla/net/NrIceStunAddrMessageUtils.h";
@@ -19,7 +19,6 @@
 using WebrtcGlobalLog from "mozilla/media/webrtc/WebrtcGlobal.h";
 using mozilla::dom::RTCIceServer from "mozilla/dom/RTCConfigurationBinding.h";
 using mozilla::dom::RTCIceTransportPolicy from "mozilla/dom/RTCConfigurationBinding.h";
-using TabId from "mozilla/dom/ipc/IdType.h";
 
 // ParamTraits stuff for our own classes
 using MediaPacket from "mtransport/mediapacket.h";
@@ -46,10 +45,7 @@
                      RTCIceServer[] iceServers,
                      RTCIceTransportPolicy icePolicy);
 
-  async SetProxyConfig(TabId id,
-                       LoadInfoArgs args, // Does this have the id?
-                       nsCString alpn,
-                       int proxyPolicy);
+  async SetProxyConfig(WebrtcProxyConfig proxyConfig);
 
   async EnsureProvisionalTransport(string transportId,
                                    string localUfrag,